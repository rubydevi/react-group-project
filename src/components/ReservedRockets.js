import React from 'react';
import { useSelector } from 'react-redux';

const ReservedRockets = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  if (reservedRockets.length === 0) {
    return <div>You have no reserved rockets.</div>;
  }

  return (
    <ul className="reserved-rockets">
      {reservedRockets.map((rocket) => (
        <li className="reserved-rocket" key={rocket.id}>{rocket.name}</li>
      ))}
    </ul>
  );
};

export default ReservedRockets;
