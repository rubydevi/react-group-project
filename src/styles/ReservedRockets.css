.reserved-rockets {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.reserved-rocket {
  padding: 3% 5%;
  border: 1px solid;
  width: 100%;
}
