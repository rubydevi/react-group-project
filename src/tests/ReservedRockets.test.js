import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ReservedRockets from '../components/ReservedRockets';

// Create a mock store with the state you want to use in the test
const mockStore = configureStore([]);

test('ReservedRockets component renders correctly with reserved rockets', () => {
  const mockReservedRockets = [
    {
      id: '5e9d0d95eda69955f709d1eb',
      name: 'Falcon 1',
      description: 'Description 1',
      flickr_images: ['https://imgur.com/DaCfMsj.jpg'],
      reserved: true,
    },
    {
      id: '5e9d0d95eda69973a809d1ec',
      name: 'Falcon 9',
      description: 'Description 2',
      flickr_images: ['https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg'],
      reserved: true,
    },
  ];

  const store = mockStore({
    rockets: {
      rockets: mockReservedRockets,
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <ReservedRockets />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ReservedRockets component renders correctly with no reserved rockets', () => {
  const mockReservedRockets = [];

  const store = mockStore({
    rockets: {
      rockets: mockReservedRockets,
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <ReservedRockets />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
