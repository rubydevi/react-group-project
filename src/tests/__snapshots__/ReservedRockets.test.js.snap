// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedRockets component renders correctly with no reserved rockets 1`] = `
<div>
  You have no reserved rockets.
</div>
`;

exports[`ReservedRockets component renders correctly with reserved rockets 1`] = `
<div
  className="list-group"
  onKeyDown={[Function]}
>
  <div
    className="list-group-item"
    data-rr-ui-event-key={null}
    onClick={[Function]}
  >
    Falcon 1
  </div>
  <div
    className="list-group-item"
    data-rr-ui-event-key={null}
    onClick={[Function]}
  >
    Falcon 9
  </div>
</div>
`;
